concordium-consensus/src/Concordium/MultiVersion.hs

```diff
@@ -403,42 +404,44 @@ newGenesis ::
-                                    mvcStateConfig
-                                    mvcRuntimeParameters
-                                    vcIndex
-                                    vcGenesisHeight
+newGenesis (PVGenesisData (gd :: GenesisData pv)) vcGenesisHeight = case sConsensusParametersVersionFor (chainParametersVersion @(ChainParametersVersionFor pv)) of
+    SConsensusParametersVersion0 ->
+        MVR $
+            \MultiVersionRunner
+                { mvCallbacks = Callbacks{..},
+                  mvConfiguration = MultiVersionConfiguration{..},
+                  ..
+                } -> do
+                    mvLog Runner LLInfo $
+                        "Starting new chain"
+                            ++ " at absolute height "
+                            ++ show vcGenesisHeight
+                    oldVersions <- readIORef mvVersions
+                    let vcIndex = fromIntegral (length oldVersions)
+                    (vcContext, st) <-
+                        runLoggerT
+                            ( Skov.initialiseNewSkov
+                                gd
+                                ( Skov.SkovConfig @pv @gsconf @finconf
+                                    ( globalStateConfig
+                                        mvcStateConfig
+                                        mvcRuntimeParameters
+                                        vcIndex
+                                        vcGenesisHeight
+                                    )
+                                    mvcFinalizationConfig
+                                    UpdateHandler
                                 )
-                                mvcFinalizationConfig
-                                UpdateHandler
                             )
-                        )
-                        mvLog
-                vcState <- newIORef st
-                let vcShutdown = Skov.shutdownSkov vcContext =<< liftIO (readIORef vcState)
-                let newEConfig :: VersionedConfiguration gsconf finconf pv
-                    newEConfig = VersionedConfiguration{..}
-                writeIORef mvVersions (oldVersions `Vec.snoc` newVersion newEConfig)
-                -- Notify the network layer we have a new genesis.
-                notifyRegenesis (Just (genesisBlockHash gd))
+                            mvLog
+                    vcState <- newIORef st
+                    let vcShutdown = Skov.shutdownSkov vcContext =<< liftIO (readIORef vcState)
+                    let newEConfig :: VersionedConfiguration gsconf finconf pv
+                        newEConfig = VersionedConfiguration{..}
+                    writeIORef mvVersions (oldVersions `Vec.snoc` newVersion newEConfig)
+                    -- Notify the network layer we have a new genesis.
+                    notifyRegenesis (Just (genesisBlockHash gd))
+    SConsensusParametersVersion1 -> error "New consensus version not implemented yet." -- TODO: implement
```

Comment:
Please make a task that points to this. FIXME's and TODOs in the code should not be there without accompanying tasks.

Reply:
Added this task. https://github.com/Concordium/concordium-node/issues/700